packages/sdk/src/adapters/multipeer/rules.ts

```diff
@@ -911,11 +912,69 @@ export const Rules: { [id in Payloads.PayloadType]: Rule } = {
     'set-sound-state': {
         ...DefaultRule,
         synchronization: {
-            stage: 'create-animations',
+            stage: 'active-sound-instances',
             before: 'ignore',
             during: 'queue',
             after: 'allow'
+        },
+        session: {
+            ...DefaultRule.session,
+            beforeReceiveFromApp: (
+                session: Session,
+                message: Message<Payloads.SetSoundState>
+            ) => {
+                const syncActor = session.actorSet[message.payload.actorId];
+                if (syncActor) {
+                    syncActor.activeSoundInstances = syncActor.activeSoundInstances || [];
+
+                    const basisTime = Date.now() / 1000.0;
+                    if (message.payload.soundCommand === SoundCommand.Start) {
+                        syncActor.activeSoundInstances.push({ message, basisTime });
+                    } else {
+                        // find the existing message that needs to be updated
+                        const activeSoundInstance = syncActor.activeSoundInstances.filter(
+                            item => item.message.payload.id === message.payload.id).shift();
+
+                        // if sound expired then skip this message completely
+                        if (!activeSoundInstance) {
+                            return undefined;
+                        }
+                        // Remove the existing sound instance (we'll add an updated one below).
+                        syncActor.activeSoundInstances =
+                            (syncActor.activeSoundInstances || []).filter(
+                                item => item.message.payload.id !== message.payload.id);
+
+                        // store the updated sound instance if sound isn't stopping
+                        if (message.payload.soundCommand !== SoundCommand.Stop) {
+
+                            // update startimeoffset and update basistime in oldmessage.
+                            const targetTime = Date.now() / 1000.0;
+                            if (activeSoundInstance.message.payload.options.paused !== false) {
```

Comment:
>activeSoundInstance.message.payload.options.paused !== false [](start = 32, length = 60)

This will return true if `paused` is undefined.

Reply:
Is that desired?

---
In reply to: [261766426](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/228#discussion_r261766426) [](ancestors = 261766426)